### PR TITLE
Assert that auth is working (positive/negative)

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
@kevinearls another one for you

File: test/generate-trace.sh [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+# span is because of the auth failure instead of network latency
+tracegen -otlp-endpoint localhost:55685 -otlp-insecure -service e2e-test-agent-without-auth &>> ./test/tracegen.log
 if [ $? != 0 ]; then
-    echo "Failed to generate a trace."
+    echo "Failed to generate a trace to the agent that does not authenticate with the collector."
     exit 1
 fi
 
-echo "✅ Trace generated."
+tracegen -otlp-endpoint localhost:55680 -otlp-insecure -service e2e-test-agent-with-auth &>> ./test/tracegen.log
```
Comment (kevinearls) [inline test/generate-trace.sh:11]: One more nit.  Can you use a different name for logs?  Otherwise we clobber the output of the first run.

**Review by kevinearls — APPROVED** [review]
Once again LGTM except with one small nit

File: test/generate-trace.sh (hunk 1 above)
Comment (jpkrohling) [reply test/generate-trace.sh]: How does it look now?

Comment (jpkrohling) [reply test/generate-trace.sh]: I'll merge it for now, but let me know if it can be improved further.

**Pull request merged**